src/docfx/build/page/AttributeTransformer.cs

```diff
@@ -47,19 +48,16 @@ JValue Load(string propertyName, string uid, Document referencedFile, Document r
                 {
                     try
                     {
-                        if (rootFile != null)
+                        if (t_recursionDetector.Value.Contains((propertyName, uid, referencedFile, rootFile)))
                         {
-                            if (t_recursionDetector.Value.Contains(new UidPropertyReference(uid, propertyName, referencedFile, rootFile), new UidPropertyReferenceComparer()))
-                            {
-                                throw Errors.CircularReference(rootFile, t_recursionDetector.Value.Select(x => x.ReferencedFile).ToList()).ToException();
-                            }
-                            t_recursionDetector.Value.Push(new UidPropertyReference(uid, propertyName, referencedFile, rootFile));
+                            throw Errors.CircularReference(rootFile, t_recursionDetector.Value.Where(x => x.rootFile == rootFile).Select(x => x.referencedFile).ToList()).ToException();
                         }
+                        t_recursionDetector.Value.Push((propertyName, uid, referencedFile, rootFile));
                         return new JValue(TransformContent(context, errors, attribute, value, file, buildChild));
                     }
                     finally
                     {
-                        if (rootFile != null && t_recursionDetector.Value.Count > 0)
+                        if (t_recursionDetector.Value.Count > 0)
```

Comment:
nit: Assert the count must > 0, so if there're some bugs in code, `Pop` will throw and let us know.